apps/nextjs/src/pages/auth/signin.tsx

```diff
@@ -1,13 +1,12 @@
 import { signIn } from "next-auth/react";
-import Image from "next/image";
-import agreetoLogo from "../../assets/icon512.png";
-
 import { Button, GoogleLogo, MicrosoftLogo, Spinner } from "@agreeto/ui";
 import { type GetServerSideProps, type NextPage } from "next";
 import { trpc } from "../../utils/trpc";
 import { Membership } from "@agreeto/api/types";
 import { useRouter } from "next/router";
 import { FaStripe } from "react-icons/fa";
+import { Card } from "../../components/card";
+import { appRouter, createContext } from "@agreeto/api";
 
 export const getServerSideProps: GetServerSideProps = async (ctx) => {
```

Comment:
just for my understanding. 

- on the extension we click on `signIn`
- opens this page here
- which then redirects to the OAuthProvider
- which contains a callback for the trial checkout page
- which calls back to this page here
?

Not sure how this flow goes

Reply:
![CleanShot 2022-11-29 at 21 06 27@2x](https://user-images.githubusercontent.com/51714798/204638046-4ba489d3-cc12-4840-aa17-0b438f848f3f.png)

Reply:
Recorded explanation. Hope it will make it clear

https://share.cleanshot.com/ShUd74